manager/integration/tests/test_basic.py

```diff
@@ -4438,3 +4439,87 @@ def test_space_usage_for_rebuilding_only_volume():  # NOQA
+    Test the failed backup would be automatically deleted.
+
+    1.   Set the default setting `backupstore-poll-interval` to 60 (seconds)
+    2.   Set the defualt setting `failed-backup-ttl` to 3 (minutes)
+    3.   Create a volume and attach to the current node
+    4.   Create a empty backup for creating the backup volume
+    5.   Write some data to the volume
+    6.   Create a backup of the volume
+    7.   Crash all replicas
+    8.   Wait and check the backup failed
+    9.   Wait and check if the backup was deleted automatically.
+    10.  Cleanup
+    """
+    failed_backup_auto_cleanup(client, core_api, volume_name, size=str(256 * Mi))
+
+
+def failed_backup_auto_cleanup(client, core_api, volume_name, size):  # NOQA
+    bt_poll_interval = "60"
+    failed_backup_ttl = "3"
+
+    # set backupstore poll interval to 60 sec
+    set_backupstore_poll_interval(client, bt_poll_interval)
+
+    # set failed backup time to live to 3 min
+    failed_backup_ttl_setting = client.by_id_setting(SETTING_FAILED_BACKUP_TTL)
+    new_failed_backup_ttl_setting = client.update(
+        failed_backup_ttl_setting, value=failed_backup_ttl)
+    assert new_failed_backup_ttl_setting.value == failed_backup_ttl
+
+    backupstore_cleanup(client)
+
+    # create a volume
+    vol = create_and_check_volume(client, volume_name,
+                                  num_of_replicas=2, size=size)
+
+    # attatch the volume
+    vol.attach(hostId=get_self_host_id())
+    vol = wait_for_volume_healthy(client, volume_name)
+
+    # create a snapshot and a successful backup
+    snap = create_snapshot(client, volume_name)
+    vol.snapshotBackup(name=snap.name)
+
+    # write some data to the volume
+    data = {
+        'pos': 0,
+        'content': common.generate_random_data(int(size)),
+    }
+    write_volume_data(vol, data)
+
+    # create a snapshot and a backup
+    snap = create_snapshot(client, volume_name)
+    vol.snapshotBackup(name=snap.name)
+
+    # check backup status is in an InProgress state
+    _, backup = find_backup(client, volume_name, snap.name)
+    backup_id = backup.id
+    def backup_inprogress_predicate(b):
+        return b.id == backup_id and "InProgress" in b.state
+    vol = common.wait_for_backup_state(client, volume_name,
+                                       backup_inprogress_predicate)
+
+    # crash all replicas of the volume
+    crash_replica_processes(client, core_api, volume_name)
+
+    # backup status should be in an Error state and with an error message
+    def backup_failure_predicate(b):
+        return b.id == backup_id and "Error" in b.state and b.error != ""
+    vol = common.wait_for_backup_state(client, volume_name,
+                                       backup_failure_predicate)
+
+    # wait in 5 minutes for automatic failed backup cleanup
+    wait_for_backup_delete(client, volume_name, backup.name)
+
+    # cleanup and reset
+    delete_backup_volume(client, volume_name)
+    cleanup_volume(client, vol)
```

Comment:
nit: already handled in [cleanup_client](https://github.com/longhorn/longhorn-tests/blob/master/manager/integration/tests/common.py#L1478-L1505).

Reply:
Removed. Thanks.